subprojects/packagefiles/cmark-gfm/meson.build

```diff
@@ -0,0 +1,73 @@
+project('cmark-gfm', 'c', version: '0.29.0.gfm.6', default_options: ['warning_level=3'])
+
+cdata = configuration_data({
+  'HAVE_STDBOOL_H': true,
+  'HAVE___BUILTIN_EXPECT': true,
+  'HAVE___ATTRIBUTE__': true,
+})
+
+configure_file(
+  format: 'cmake@',
+  input: 'src/config.h.in',
+  output: 'config.h',
+  configuration: cdata)
+
+varr = meson.project_version().split('.')
+vdata = configuration_data({
+  'PROJECT_VERSION_MAJOR' : varr[0],
+  'PROJECT_VERSION_MINOR' : varr[1],
+  'PROJECT_VERSION_PATCH' : varr[2],
+  'PROJECT_VERSION_GFM' : varr[4],
+})
+
+configure_file(
+  format: 'cmake@',
+  input: 'src/cmark-gfm_version.h.in',
+  output: 'cmark-gfm_version.h',
+  configuration: vdata)
+
+src = [
+  'src/cmark.c',
+  'src/arena.c',
+  'src/footnotes.c',
+  'src/linked_list.c',
+  'src/map.c',
+  'src/plaintext.c',
+  'src/plugin.c',
+  'src/registry.c',
+  'src/syntax_extension.c',
+  'src/node.c',
+  'src/iterator.c',
+  'src/blocks.c',
+  'src/inlines.c',
+  'src/scanners.c',
+  'src/utf8.c',
+  'src/buffer.c',
+  'src/references.c',
+  'src/render.c',
+  'src/man.c',
+  'src/xml.c',
+  'src/html.c',
+  'src/commonmark.c',
+  'src/latex.c',
+  'src/houdini_href_e.c',
+  'src/houdini_html_e.c',
+  'src/houdini_html_u.c',
+  'src/cmark_ctype.c',
+  'extensions/autolink.c',
+  'extensions/core-extensions.c',
+  'extensions/ext_scanners.c',
+  'extensions/ext_scanners.h',
+  'extensions/strikethrough.c',
+  'extensions/table.c',
+  'extensions/tagfilter.c',
+  'extensions/tasklist.c',
+]
+
+# Many unused parameters warnings when these warnings are enabled.
+add_project_arguments('-fvisibility=hidden', language: 'c')
```

Comment:
IIRC this one can be done with gnu_symbol_visibility.

Reply:
Does the symbol table of the shared library expose a bunch of private symbols?

Given there's an export file, I imagine the source code annotates the functions which are public, so specifying `gnu_symbol_visibility: 'hidden'` is beneficial and cleans the symbol table while also allowing the compiler to generate slightly more optimized code.

Reply:
Given that this is being built as a static library I don't think visibility actually affects anything. If it were a shared library then the visibility attribute would have a purpose.